### PR TITLE
Build type updates

### PR DESCRIPTION
**Pull request by amadio**

### CODE REVIEW

File: cmake/modules/SetUpLinux.cmake [hunk 1]
```diff
@@ -108,16 +108,10 @@ if(CMAKE_COMPILER_IS_GNUCXX)
...
-  set(CMAKE_CXX_FLAGS_OPTIMIZED      "-Ofast -DNDEBUG"  CACHE STRING "Flags for a highly optimised build")
   set(CMAKE_CXX_FLAGS_DEBUG          "-g"               CACHE STRING "Flags for a debug build")
-  set(CMAKE_CXX_FLAGS_DEBUGFULL      "-g3"              CACHE STRING "Flags for a debug build with more debug info")
-  set(CMAKE_CXX_FLAGS_PROFILE        "-g3 -ftest-coverage -fprofile-arcs" CACHE STRING "")
   set(CMAKE_C_FLAGS_RELWITHDEBINFO   "-O3 -g -DNDEBUG"  CACHE STRING "Flags for release build with debug info")
   set(CMAKE_C_FLAGS_RELEASE          "-O3 -DNDEBUG"     CACHE STRING "Flags for release build")
-  set(CMAKE_C_FLAGS_OPTIMIZED        "-Ofast -DNDEBUG"  CACHE STRING "Flags for a highly optimised build")
   set(CMAKE_C_FLAGS_DEBUG            "-g"               CACHE STRING "Flags for a debug build")
-  set(CMAKE_C_FLAGS_DEBUGFULL        "-g3 -fno-inline"  CACHE STRING "Flags for a debug build with more debug info")
-  set(CMAKE_C_FLAGS_PROFILE          "-g3 -fno-inline -ftest-coverage -fprofile-arcs" CACHE STRING "")
```
Comment (pcanal) [inline cmake/modules/SetUpLinux.cmake]: The original idea of this flags was to (I guess 'eventually') have it know/set the compiler depend profiling options.

File: cmake/modules/SetUpLinux.cmake [hunk 2]
```diff
@@ -108,16 +108,10 @@ if(CMAKE_COMPILER_IS_GNUCXX)
   # Select flags.
   set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DNDEBUG"  CACHE STRING "Flags for release build with debug info")
   set(CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG"     CACHE STRING "Flags for release build")
-  set(CMAKE_CXX_FLAGS_OPTIMIZED      "-Ofast -DNDEBUG"  CACHE STRING "Flags for a highly optimised build")
   set(CMAKE_CXX_FLAGS_DEBUG          "-g"               CACHE STRING "Flags for a debug build")
-  set(CMAKE_CXX_FLAGS_DEBUGFULL      "-g3"              CACHE STRING "Flags for a debug build with more debug info")
-  set(CMAKE_CXX_FLAGS_PROFILE        "-g3 -ftest-coverage -fprofile-arcs" CACHE STRING "")
```
Comment (pcanal) [inline cmake/modules/SetUpLinux.cmake]: The original idea of this flags was to (I guess 'eventually') have it know/set the compiler depend profiling options.

File: cmake/modules/SetUpLinux.cmake (hunk 1 above)
Comment (amadio) [reply cmake/modules/SetUpLinux.cmake]: Sure, but it's redundant with the `coverage` option, and it sets GCC flags for everyone, which is obviously wrong. That's why I'm removing it. I don't plan to merge this before discussing with more people to check that these are really not in use, though.

**Comment by amadio** [comment]
@oshadura Is any of the build types being removed used by your builds (i.e. modules, rootbench, etc?)

File: cmake/modules/SetUpLinux.cmake (hunk 1 above)
Comment (pcanal) [reply cmake/modules/SetUpLinux.cmake]: >  and it sets GCC flags for everyone, which is obviously wrong.

Yes any such flag that only flag for a subset of platform ought to error out and/or warn on the other platforms.

I agree that the 'coverage' variable is a better solution that a build type ... albeit it is currently suffering the same weakness (only works for GCC) :)

**Comment by oshadura** [comment]
@amadio thank you for asking!For me everything is fine, please go ahead!

**Comment by amadio** [comment]
The failure is from the race condition, which was reverted. I'll re-run Jenkins and merge.

**Comment by amadio** [comment]
@phsft-bot build just on ROOT-performance-centos7-multicore/default

File: cmake/modules/SetUpLinux.cmake (hunk 1 above)
Comment (amadio) [reply cmake/modules/SetUpLinux.cmake]: Ok, I will get the `coverage` option fixed in another pull request after I merge this.

**Pull request merged**